repositories/slot_repository.go

```diff
@@ -0,0 +1,24 @@
+package repositories
+
+import (
+	"airport-app-backend/models"
+)
+
+type ISlotRepository interface {
+	RetrieveAllSlots(int, bool) ([]models.Slot, error)
+}
+
+func (sr *ServiceRepository) RetrieveAllSlots(page int, status bool) ([]models.Slot, error) {
+	var slots []models.Slot
+
+	query := sr.db.Offset(page * DEFAULT_PAGE_SIZE).Limit(DEFAULT_PAGE_SIZE)
+	if !status {
+		query = query.Where("status IN (?, ?)", "Booked", "Reserved")
+	} else {
+		query = query.Where("status = ?", "Available")
+	}
+	if err := query.Find(&slots).Error; err != nil {
```

Comment:
assignment and conditionals in the same line (though supported in go) are confusing. please separate out into 2 lines